src/object/rpc_csum.c

```diff
@@ -182,15 +182,17 @@ crt_proc_struct_dcs_iod_csums_adv(crt_proc_t proc, crt_proc_op_t proc_op,
 
 	if (DECODING(proc_op)) {
 		PROC(uint32_t, &iod_csum->ic_nr);
-		D_ALLOC_ARRAY(iod_csum->ic_data, iod_csum->ic_nr);
```

Comment:
@ryon-jensen can you please explain how we can get into a situation where ic_nr is 0 here?
and why we rely on ic_data being not-NULL if ic_nr is 0. that sounds like a bug TBH. i attempted at fixing it here and in src/common/checksum.c; but not sure if there are other places/

Reply:
ic_nr can be 0 when the iod.iod_size is 0. I don't think ic_data needs to be not-NULL necessarily, so this change makes sense.